atlassian_jwt_auth/frameworks/flask/backend.py

```diff
@@ -0,0 +1,41 @@
+from flask import Response, current_app, g, request as current_req
+
+from ..common.backend import Backend
+from ..common.utils import SettingsDict
+
+
+class FlaskBackend(Backend):
+    def get_authorization_header(self, request=None):
+        if request is None:
+            request = current_req
```

Comment:
What if this is still `None`?

Reply:
If that happens, one of the fundamental concepts in Flask is broken since current_req is a global value set by Flask that always contains the current request who's lifecycle you are inside of.

To be exact, if this is still `None`, you're not using it inside of a Flask request context and you'll get exceptions.

In reality, as long as you are only using the FlaskBackend with Flask, that simply won't happen.

The only reason this `if` is here at all instead of simply using `current_req` is because I wanted to allow us to pass in an override request during testing.